code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -19,7 +19,18 @@
 
 	if(href_list["PRG_openfile"])
 		. = 1
-		open_file = href_list["PRG_openfile"]
+		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
+		var/datum/computer_file/file = HDD.find_file_by_name(href_list["PRG_openfile"])
+		var/checkpass = ""
+		var/currentpass = file.password
```

Comment:
Unsafe access of file.
This whole password check should be put inside `ui_interact()` around line 225

Reply:
Do I make a proc to check for the password?

Reply:
You should make password protection validation a proc, yes. This method of doing:
```dm
		if (!currentpass)
			// do one thing
		else
			// check password and do the other
```

is bad. Very bad. A file should have a proc `proc/can_modify_file(mob/user, prompt_for_password = FALSE, input_password = "")`

Rough contents would look like this:
```dm
/datum/computer_file/proc/can_modify_file(mob/user, prompt_for_password = FALSE, input_password = "")
	if (!password)
		return TRUE

	if (prompt_for_password)
		if (!ismob(user))
			return FALSE

		input_password = sanitize(input(user, "Please enter a password to access file '[name]':"))

	return password == input_password
```

This allows for 100x cleaner functionality:
```dm
		if (!file.can_access_file(usr, TRUE))
			return 1

		open_file = href_list["PRG_openfile"]
```